src/client/components/forms/userCollection.js

```diff
@@ -0,0 +1,306 @@
+				name: ''
+			};
+		}
+		else {
+			newCollaborator = newCollab;
+		}
+		const newCollaborators = this.state.collection.collaborators;
+		newCollaborators[index] = newCollaborator;
+		this.setState(prevState => ({
+			collection: {
+				...prevState.collection,
+				collaborators: newCollaborators
+			},
+			errorText: prevState.errorText
+		}));
+	}
+
+	render() {
+		if (this.state.collection.collaborators.length === 0) {
+			this.handleAddCollaborator();
+		}
+
+		const privacyOptions = ['Private', 'Public'].map((option) => ({
+			name: option
+		}));
+		const entityTypeOptions = ['Author', 'Work', 'Edition', 'Edition-Group', 'Publisher'].map((entity) => ({
+			name: entity
+		}));
+		const initialName = this.state.collection.name;
+		const initialDescription = this.state.collection.description;
+		const initialPrivacy = this.state.collection.public ? 'Public' : 'Private';
+		const initialType = this.state.collection.entityType;
+		const {errorText} = this.state;
+		const errorAlertClass =
+			classNames('text-center', 'margin-top-1', {hidden: !errorText});
+		const submitLabel = this.props.collection.name ? 'UPDATE' : 'CREATE';
+
+		/* eslint-disable react/jsx-no-bind */
+		return (
+			<Grid>
+				<h1>Create a collection</h1>
+				<Row>
+					<Col md={12}>
+						<p className="lead">Create Your Collection</p>
+					</Col>
+				</Row>
+				<Row>
+					<Col
+						id="collectionForm"
+						md={8}
+						mdOffset={2}
+					>
+						<form
+							className="form-horizontal"
+							onSubmit={this.handleSubmit}
+						>
+							<CustomInput
+								defaultValue={initialName}
+								label="Name"
+								ref={(ref) => this.name = ref}
+								type="text"
+							/>
+							<CustomInput
+								defaultValue={initialDescription}
+								label="Description"
+								ref={(ref) => this.description = ref}
+								type="textarea"
+							/>
+							<SelectWrapper
+								base={ReactSelect}
+								defaultValue={initialType}
+								disabled={!this.props.canEditType}
+								idAttribute="name"
+								label="Entity Type"
+								labelAttribute="name"
+								options={entityTypeOptions}
+								placeholder="Select Entity Type"
+								ref={(ref) => this.entityType = ref}
+							/>
+							<SelectWrapper
+								base={ReactSelect}
+								defaultValue={initialPrivacy}
+								idAttribute="name"
+								label="Privacy"
+								labelAttribute="name"
+								options={privacyOptions}
+								placeholder="Select Privacy"
+								ref={(ref) => this.privacy = ref}
+							/>
+							{
+								this.state.collection.collaborators.map((collaborator, index) => (
+									<div key={collaborator.id}>
+										<EntitySearchFieldOption
+											instanceId="collaboratorSearchField"
+											label="Select Collaborator"
+											name="editor"
+											type="editor"
+											value={collaborator}
+											onChange={(newCollaborator) => this.handleChangeCollaborator(newCollaborator, index)}
+										/>
+										<Button
```

Comment:
Putting the "remove" buttons on the same line as the collaborator select makes it clear that they are related. You can pass a `buttonAfter` property, which with a small modification to the EntitySearchFieldOption component will be [passed down to CustomInput](https://github.com/bookbrainz/bookbrainz-site/blob/77475b87fe748aecbb5202bc6774188b251051d8/src/client/input.js#L124)

The modification is passing down the props to CustomInput [here](https://github.com/bookbrainz/bookbrainz-site/blob/77475b87fe748aecbb5202bc6774188b251051d8/src/client/entity-editor/common/entity-search-field-option.js#L108): 
```
<CustomInput label={labelElement} tooltipText={this.props.tooltipText}>
<CustomInput label={labelElement} tooltipText={this.props.tooltipText} {...this.props}>
```

With that set up, you can do the following:
```
this.state.collection.collaborators.map((collaborator, index) => {
								const buttonAfter = (
									<Button
										bsSize="small"
										bsStyle="danger"
										type="button"
										onClick={() => this.handleRemoveCollaborator(index)}
									>
										Remove
									</Button>
								);
								return (
									<div key={collaborator.id}>
										<EntitySearchFieldOption
											buttonAfter={buttonAfter}
											instanceId="collaboratorSearchField"
											label="Select Collaborator"
											name="editor"
											type="editor"
											value={collaborator}
											onChange={(newCollaborator) => this.handleChangeCollaborator(newCollaborator, index)}
										/>
									</div>
								);
							})
```

<img width="687" alt="Capture d’écran 2020-06-03 à 16 17 09" src="https://user-images.githubusercontent.com/6179856/83647893-b60a1880-a5b5-11ea-83af-3e2ac5a6a8a7.png">

Reply:
Thank you for thiss. I forgot about `buttonAfter` I was trying to adjust the button to the right side by `col-md-10`  `col-md-2`, but that way it was looking very bad